src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/QueryExecutor.java

```diff
@@ -120,13 +124,25 @@ protected Uni<Optional<Schema.CqlTable>> getSchema(String namespace, String coll
     final Uni<Schema.CqlKeyspaceDescribe> cqlKeyspaceDescribeUni =
         stargateRequestInfo.getStargateBridge().describeKeyspace(describeKeyspaceQuery);
     return cqlKeyspaceDescribeUni
-        .onItem()
-        .transform(
-            cqlKeyspaceDescribe -> {
+        .onItemOrFailure()
+        .transformToUni(
+            (cqlKeyspaceDescribe, error) -> {
+              if (error != null
+                  && (error instanceof StatusRuntimeException sre
+                      && sre.getStatus().getCode() == Status.Code.NOT_FOUND)) {
+                return Uni.createFrom()
+                    .failure(
+                        new RuntimeException(
+                            new JsonApiException(
+                                ErrorCode.NAMESPACE_DOES_NOT_EXIST,
+                                "The provided namespace does not exist: " + namespace)));
```

Comment:
Is this meant to improve the following error that happens when you call `find()` on a collection that doesn't exist?

```
Error: Command "find" failed with the following errors: [{"message":"INVALID_ARGUMENT: table articles22 does not exist","exceptionClass":"StatusRuntimeException"}]

```

Reply:
@vkarpov15 I think this is about `namespace`, not collection (table)

Reply:
This means to improve the error when the namespace doesn't exist, not the collection. I guess you got the error when you input the wrong collection name (or `table` showed in the error message).